src/components/ActionBar.tsx

```diff
@@ -0,0 +1,155 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { RelayingServices, SmartWallet } from '@rsksmart/rif-relay-sdk';
+import { SmartWalletWithBalance } from 'src/types';
+import Utils, { TRIF_PRICE } from 'src/Utils';
+import 'src/components/ActionBar.css';
+import { Col, Row, Button, Icon } from 'react-materialize';
+import AllowedTokens from 'src/components/AllowedTokens';
+
+type ActionBarProps = {
+    smartWallets: SmartWalletWithBalance[];
+    setSmartWallets: Dispatch<SetStateAction<SmartWalletWithBalance[]>>;
+    connected: boolean;
+    account?: string;
+    provider?: RelayingServices;
+    setShow: Dispatch<SetStateAction<boolean>>;
+    token: string;
+    updateInfo: boolean;
+    setToken: Dispatch<SetStateAction<string>>;
+    tokenSymbol: string;
+    setTokenSymbol: Dispatch<SetStateAction<string>>;
+};
+
+function ActionBar(props: ActionBarProps) {
+    const {
+        smartWallets,
+        setSmartWallets,
+        connected,
+        account,
+        provider,
+        setShow,
+        token,
+        updateInfo,
+        setToken,
+        tokenSymbol,
+        setTokenSymbol
+    } = props;
+
+    const [workerBalance, setWorkerBalance] = useState('0');
+
+    const setBalance = async (
+        smartWallet: SmartWallet,
+        symbol: string
+    ): Promise<SmartWalletWithBalance> => {
+        const balance = await Utils.tokenBalance(smartWallet.address, token);
+        const rbtcBalance = await Utils.getBalance(smartWallet.address);
+        const swWithBalance = {
+            ...smartWallet,
+            balance: `${Utils.fromWei(balance)} ${symbol}`,
+            rbtcBalance: `${Utils.fromWei(rbtcBalance)} RBTC`
+        };
+        return swWithBalance;
+    };
+
+    useEffect(() => {
+        if (!account || !provider || !token) {
+            return;
+        }
+        (async () => {
+            let index: number = 0;
+            let found: boolean = true;
+            const tempSmartWallets: SmartWalletWithBalance[] = [];
+            const symbol = await Utils.tokenSymbol(token);
+            setTokenSymbol(symbol);
+            while (found) {
```

Comment:
I think we should store the SW created into the localStorage and eventually allow the user to import a new SW. WDYT?
It may be a strong improvement because when we try to load the SWs in the testnet deployment, the step is quite slow.
We could do this in a different task.

Reply:
I like that, but we need to keep this as an initializer and if the sw is not found in the localStorage; do this since we can try to use it in a different computer.

Reply:
Yes, or ask the users to specify the address of the SW they want to import, without guessing it. I created the [related task](https://rsklabs.atlassian.net/browse/PP-204).